macos/Sources/Features/Terminal/TerminalManager.swift

```diff
@@ -125,6 +125,12 @@ class TerminalManager {
     }
 
     private func newTab(to parent: NSWindow, withBaseConfig base: Ghostty.SurfaceConfiguration?) {
+        // If the parent window is a QuickTerminalWindow, we early return with an alert.
+        if let controller = parent.windowController as? QuickTerminalController {
```

Comment:
I think we should move this logic out to the `onNewTab` notification and instead of binding us to QuickTerminalController instead test that the controller is a `TerminalController` and ignore all else. This way if we introduce more terminal types then the manager only responds to that.